packages/ember-data/lib/system/record_array_manager.js

```diff
@@ -261,7 +261,7 @@ export default Ember.Object.extend({
   willDestroy: function(){
     this._super();
 
-    forEach(flatten(values(this.filteredRecordArrays.values)), destroy);
+    forEach(flatten(values(this.filteredRecordArrays._values)), destroy);
```

Comment:
Fair question! That would probably render this polyfill unnecessary, correct?

I'm sure there's a reason why that's not used here, but maybe it could be rewritten to do so?